views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js

```diff
@@ -161,18 +162,59 @@ define([
         $score[0].dataset.for = id;
         $input[0].dataset.index = id;
 
+        if (this.inGapMode() === true) {
+            this.emptyGapFields();
+            const gapExpression = interaction.prop('gapExpression');
+            const gapCount = (gapExpression.match(/\\taoGap/g) || []).length;
+            if (gapCount > 0) {
+                newCorrectAnswer = [];
+                for (let i = 0; i < gapCount; i++) {
+                    newCorrectAnswer.push(' ');
+                }
+                newCorrectAnswer = newCorrectAnswer.join(',');
+            } else {
+                newCorrectAnswer = '';
+            }
+        } else {
+            newCorrectAnswer = '';
+            this.toggleResponseMode(false);
+        }
+
         let existingReponses = this.getExistingCorrectAnswerOptions();
         const response = interaction.getResponseDeclaration();
-        const correctResponse = response.getCorrect();
-        if (!!correctResponse) {
-            let correctResponseIndex = existingReponses.indexOf(correctResponse[0]);
-            if (correctResponseIndex !== -1) {
-                existingReponses.splice(correctResponseIndex, 1);
-                existingReponses.unshift(correctResponse[0]);
+        if (!!existingReponses.length) {
+            this.emptyGapFields();
+            const gapExpression = interaction.prop('gapExpression');
+            const gapCount = (gapExpression.match(/\\taoGap/g) || []).length;
+            if (this.inGapMode() === true) {
+                if (!gapCount) {
+                    response.removeMapEntries();
+                    response.setCorrect(null);
+                    $score[0].value = defaultScore;
+                    const inputValue = this.checkValues(id);
+                    this.correctResponses.set(id, Object.assign(inputValue, { response: newCorrectAnswer }));
+                    return false;
+                }
+            } else {
+                if (!!gapCount) {
```

Comment:
why we checking here gapCount if it is not GapMode?

Reply:
I'm checking here the case when the user has stitched between gap mode on/off, but he didn't add any gap. The map entries are removed when the author adds a gap to the field. So I need to force the behaviour if the author switched to gap mode on/off but did not click on the gap button.